src/ui/public/courier/fetch/strategy/__tests__/search.js

```diff
@@ -1,118 +0,0 @@
-import _ from 'lodash';
-import expect from 'expect.js';
-import ngMock from 'ng_mock';
-
-import { SearchStrategyProvider } from '../search';
-import StubIndexPatternProvider from 'test_utils/stub_index_pattern';
-
-describe('SearchStrategyProvider', () => {
-  require('test_utils/no_digest_promises').activateForSuite();
-
-  let search;
-  let reqsFetchParams;
-  let IndexPattern;
-
-  beforeEach(ngMock.module('kibana'));
-
-  beforeEach(ngMock.inject((Private) => {
-    search = Private(SearchStrategyProvider);
-    IndexPattern = Private(StubIndexPatternProvider);
-
-    reqsFetchParams = [
-      {
-        index: ['logstash-123'],
-        type: 'blah',
-        search_type: 'blah2',
-        body: { foo: 'bar', $foo: 'bar' }
-      }
-    ];
-  }));
-
-  describe('#clientMethod', () => {
-    it('is msearch', () => {
-      expect(search.clientMethod).to.equal('msearch');
-    });
-  });
-
-  describe('#reqsFetchParamsToBody()', () => {
-    it('filters out any body properties that begin with $', () => {
```

Comment:
interestingly, the test I added passed.  i believe it's because of the `toJson(body, JSON.stringify);` portion - that removes the `$` part.  It does skip parsing out $ for this part:
```
{
          index,
          type: fetchParams.type,
          search_type: fetchParams.search_type,
          ignore_unavailable: true,
          preference: sessionId,
        }
```

But it seems like that would never have a $ so we are safe there.

Reply:
Ah, I missed us using the `toJson` for the request body itself. I stand corrected :)